src/main/java/com/meltmedia/jgroups/aws/TagsUtils.java

```diff
@@ -52,6 +53,26 @@ public TagsUtils(final AmazonEC2 ec2, final InstanceIdentity instanceIdentity, f
   }
 
   /**
+   * If require_tags=true, configured tags will be validated against the instance tags.
+   * If one or more tags are missing on the instance, an exception will be thrown.
+   *
+   * @throws IllegalStateException
+   */
+  public TagsUtils validateTags(boolean requireTags) {
+    if(requireTags) {
+      final List<String> instanceTags = getInstanceTags().stream()
+          .map(Tag::getKey)
+          .collect(Collectors.toList());
+
+      getAwsTagNames().map(List::stream).orElse(Stream.empty())
+          .filter(configuredTag -> !instanceTags.contains(configuredTag))
+          .findFirst()
+          .ifPresent(missingTag -> {throw new IllegalStateException("instance is missing tag: " + missingTag);});
+    }
```

Comment:
Here, it might be better to map the content of the optional and then filter empty lists, then we could notify on all missing tags.

```
getAwsTagNames()
  .map(list->list.stream().filter(...).collect(...))
  .filter(list->!list.isEmpty())
  .ifPresent(allMissingTags->...)
```

Reply:
good point :-)